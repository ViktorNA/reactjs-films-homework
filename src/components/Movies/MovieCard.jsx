import propTypes from 'prop-types';
import React, { Component } from 'react';
import styles from './MovieCard.scss';

export default class MovieCard extends Component {
  state = {
    infoType: 'short',
  }

  getBackStyle() {
    const { infoType } = this.state;
    return infoType === 'short' ? styles.backShort : styles.backLong;
  }

  getCloseStyle() {
    const { infoType } = this.state;
    return infoType === 'long' ? styles.closeWrapper : styles.hiddenDesc;
  }

  getWatchStyle() {
    const { infoType } = this.state;
    return infoType === 'long' ? styles.watchNowButton : styles.hiddenDesc;
  }

  getHeadStyle() {
    const { infoType } = this.state;
    return infoType === 'video' ? styles.shortHead : styles.head;
  }

  getGenresList = genres => [...genres].slice(0, 3).join(', ');

  setInfoType = (event) => {
    this.setState({ infoType: event.target.name });
  }

  showInfo(description) {
    const { infoType } = this.state;
    switch (infoType) {
      case 'video': return (
        <div className={styles.watchWrapper}>
          <div className={styles.circle}><div className={styles.triangle} /></div>
          <div className={styles.label}>Watch Now</div>
          <button name="long" type="button" onClick={this.setInfoType} className={styles.viewInfoButton}>
            View Info
          </button>
        </div>
      );
      case 'short': return (
        <div />
      );
      default: break;
    }
    return (
      <div className={styles.visibleDesc}>
        {description}
      </div>
    );
  }

  render() {
    const {
      movie: {
        name,
        description,
        imageUrl,
        rating,
        genres,
      },
    } = this.props;
    return (
      <div className={styles.card}>
        <img className={styles.imgWrapper} src={imageUrl} alt="Movie log" />
        <div className={this.getBackStyle()}>
          <div className={this.getCloseStyle()}>
            <button name="short" type="button" className={styles.close} onClick={this.setInfoType}>x</button>
          </div>
          <div className={this.getHeadStyle()}>
            <button name="long" className={styles.movieName} type="button" onClick={this.setInfoType}>
              {name}
            </button>
            <div className={styles.ratingBox}>
              {rating}
            </div>
            <div className={styles.genresBox}>
              {this.getGenresList(genres)}
            </div>
          </div>
          {this.showInfo(description)}
          <button name="video" type="button" className={this.getWatchStyle()} onClick={this.setInfoType}>
            Watch Now
          </button>
        </div>
      </div>
    );
  }
}

MovieCard.propTypes = {
  movie: propTypes.shape({
    name: propTypes.string.isRequired,
    description: propTypes.string.isRequired,
    imageUrl: propTypes.string,
    rating: propTypes.number.isRequired,
    genres: propTypes.array,
  }),
};

MovieCard.defaultProps = {
  movie: {
    name: 'error',
    description: 'error',
    rating: 0,
    genres: ['uncnown genre'],
  },
};
