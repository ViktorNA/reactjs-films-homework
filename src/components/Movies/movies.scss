.item {
  display: grid;
  grid-gap: 20px;
  grid-template-columns: repeat(auto-fit, minmax(250px, 1fr));
  justify-items: center;
}

.loading {
  color: rgba(36, 43, 45, 0.4);
  font-size: 11px;
  text-align: center;
  margin-top: 10px;
}

.spinner {
  text-align: center;
  margin-top: 60px;
}

img {
  width: 50px;
  height: 50px;
}
