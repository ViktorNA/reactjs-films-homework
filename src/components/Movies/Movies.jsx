import 'babel-polyfill';
import React, { Component } from 'react';
import MovieCard from '../MovieCard';
import styles from './movies.scss';
import loadGif from '../../assets/images/spinner.gif';

const movies = require('../../assets/json/popular.json');

export default class Movies extends Component {
  state = { activeId: -1 };

  updateData = (value) => {
    this.setState({ activeId: value });
  };

  render() {
    const { activeId } = this.state;
    return (
      <div className={styles.wrapper}>
        <div className={styles.item}>
          {movies.results.map(movie => (
            <MovieCard
              movie={movie}
              key={movie.id}
              id={movie.id}
              activeId={activeId}
              onShowInfo={this.updateData}
            />
          ))}
        </div>
        <div className={styles.spinner}>
          <img src={loadGif} alt="loading" />
        </div>
        <div className={styles.loading}>
          LOADING
        </div>
      </div>
    );
  }
}
