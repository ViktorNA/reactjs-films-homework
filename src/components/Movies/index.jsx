import 'babel-polyfill';
import React from 'react';
import Movies from './Movies';
import styles from './index.scss';

const MovieList = () => (
  <div className={styles.wrapper}>
    <Movies />
  </div>);

export default MovieList;
