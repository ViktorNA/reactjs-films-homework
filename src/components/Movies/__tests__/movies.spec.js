import TestRenderer from 'react-test-renderer';
import React from 'react';
import Movies from '../Movies';

describe('Snapshot test', () => {
  it('works correctly', () => {
    const testRender = TestRenderer.create(<Movies />);
    expect(testRender.toJSON()).toMatchSnapshot();
  });
});
