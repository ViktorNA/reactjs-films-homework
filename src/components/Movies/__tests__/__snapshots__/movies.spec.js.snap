// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test works correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="item"
  >
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//AnLCqcH66Ecu8VFmXu3RofK4SMK.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          T-34
        </div>
        <div
          className="ratingBox"
        >
          4.5
        </div>
        <div
          className="genresBox"
        >
          War, Drama, Adventure
        </div>
        <div
          className="hidden"
        >
          1941 - WWii. the second lieutenant Nikolai ivushkin, commander of a t-34, engages in an unequal battle against the tank ace Klaus Jager in a battle near moscow. His mission is more of a suicide - to destroy a dozen german tanks, all by himself. that said, luck does favour the bold. He wins the battle, barely survives, but loses his tank and lands himself in captivity for three long years... there was little to no chance for ivushkin and Jager to meet again, but the war knows how to throw a curve ball.in the spring of 1944, the Wehrmacht commands Jager to take charge of the ohrdruf  re range and turn it into a training center for elite german armored forces, using the latest t-34 as a running target. this is how Jager and ivushkin cross paths again. Jager o ers ivushkin to become the commander of a legendary tank and pick his crew from fellow camp prisoners. Nothing goes according to plan, though, when ivushkin uses exercises for a daring and carefully planned escape.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//svIDTNUoajS8dLEo7EosxvyAsgJ.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Glass
        </div>
        <div
          className="ratingBox"
        >
          6.8
        </div>
        <div
          className="genresBox"
        >
          Thriller, Mystery, Drama
        </div>
        <div
          className="hidden"
        >
          In a series of escalating encounters, security guard David Dunn uses his supernatural abilities to track Kevin Wendell Crumb, a disturbed man who has twenty-four personalities. Meanwhile, the shadowy presence of Elijah Price emerges as an orchestrator who holds secrets critical to both men.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//AiRfixFcfTkNbn2A73qVJPlpkUo.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Robin Hood
        </div>
        <div
          className="ratingBox"
        >
          5.8
        </div>
        <div
          className="genresBox"
        >
          Adventure, Action, Thriller
        </div>
        <div
          className="hidden"
        >
          A war-hardened Crusader and his Moorish commander mount an audacious revolt against the corrupt English crown.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Bohemian Rhapsody
        </div>
        <div
          className="ratingBox"
        >
          8.2
        </div>
        <div
          className="genresBox"
        >
          Drama, Music
        </div>
        <div
          className="hidden"
        >
          Singer Freddie Mercury, guitarist Brian May, drummer Roger Taylor and bass guitarist John Deacon take the music world by storm when they form the rock 'n' roll band Queen in 1970. Hit songs become instant classics. When Mercury's increasingly wild lifestyle starts to spiral out of control, Queen soon faces its greatest challenge yet â€“ finding a way to keep the band together amid the success and excess.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//l76Rgp32z2UxjULApxGXAPpYdAP.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Overlord
        </div>
        <div
          className="ratingBox"
        >
          6.9
        </div>
        <div
          className="genresBox"
        >
          Horror, War, Action
        </div>
        <div
          className="hidden"
        >
          France, World War II, June 1944. On the eve of D-Day, some American paratroopers fall behind enemy lines after their plane crashes during a mission consisting of destroying a German radio tower in Cielblanc, a small village near the beaches of Normandy. After reaching their target, the surviving paratroopers realize that, in addition to fighting Nazi troops that patrol the village, they also must fight against something else.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//8yZAx7tlKRZIg7pJfaPhl00yHIQ.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Escape Room
        </div>
        <div
          className="ratingBox"
        >
          6.3
        </div>
        <div
          className="genresBox"
        >
          Horror, Thriller, Action
        </div>
        <div
          className="hidden"
        >
          Six strangers find themselves in circumstances beyond their control, and must use their wits to survive.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//5Kg76ldv7VxeX9YlcQXiowHgdX6.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Aquaman
        </div>
        <div
          className="ratingBox"
        >
          6.8
        </div>
        <div
          className="genresBox"
        >
          Action, Fantasy, Science Fiction
        </div>
        <div
          className="hidden"
        >
          Once home to the most advanced civilization on Earth, the city of Atlantis is now an underwater kingdom ruled by the power-hungry King Orm. With a vast army at his disposal, Orm plans to conquer the remaining oceanic people -- and then the surface world. Standing in his way is Aquaman, Orm's half-human, half-Atlantean brother and true heir to the throne. With help from royal counselor Vulko, Aquaman must retrieve the legendary Trident of Atlan and embrace his destiny as protector of the deep.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Mortal Engines
        </div>
        <div
          className="ratingBox"
        >
          6
        </div>
        <div
          className="genresBox"
        >
          Science Fiction, Action, Adventure
        </div>
        <div
          className="hidden"
        >
          Set in a world many thousands of years in the future. Earthâ€™s cities now roam the globe on huge wheels, devouring each other in a struggle for ever diminishing resources. On one of these massive Traction Cities, Tom Natsworthy has an unexpected encounter with a mysterious young woman from the Outlands who will change the course of his life forever.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//v3QyboWRoA4O9RbcsqH8tJMe8EB.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Creed II
        </div>
        <div
          className="ratingBox"
        >
          6.7
        </div>
        <div
          className="genresBox"
        >
          Drama, Action
        </div>
        <div
          className="hidden"
        >
          Follows Adonis Creed's life inside and outside of the ring as he deals with new found fame, issues with his family, and his continuing quest to become a champion.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//ij0xoc13hGhrYIlXGGuPXWTh3vi.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          How to Train Your Dragon: The Hidden World
        </div>
        <div
          className="ratingBox"
        >
          8.1
        </div>
        <div
          className="genresBox"
        >
          Animation, Family, Adventure
        </div>
        <div
          className="hidden"
        >
          As Hiccup fulfills his dream of creating a peaceful dragon utopia, Toothlessâ€™ discovery of an untamed, elusive mate draws the Night Fury away.  When danger mounts at home and Hiccupâ€™s reign as village chief is tested, both dragon and rider must make impossible decisions to save their kind.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//wrFpXMNBRj2PBiN4Z5kix51XaIZ.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          A Star Is Born
        </div>
        <div
          className="ratingBox"
        >
          7.4
        </div>
        <div
          className="genresBox"
        >
          Drama, Music, Romance
        </div>
        <div
          className="hidden"
        >
          Seasoned musician Jackson Maine discovers â€” and falls in love with â€” struggling artist Ally. She has just about given up on her dream to make it big as a singer â€” until Jack coaxes her into the spotlight. But even as Ally's career takes off, the personal side of their relationship is breaking down, as Jack fights an ongoing battle with his own internal demons.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//t0idiLMalKMj2pLsvqHrOM4LPdQ.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          The Mule
        </div>
        <div
          className="ratingBox"
        >
          6.3
        </div>
        <div
          className="genresBox"
        >
          Crime, Drama, Thriller
        </div>
        <div
          className="hidden"
        >
          A 90 year old horticulturalist and Korean War veteran is caught transporting $3 million worth of cocaine through Illinois for a Mexican drug cartel.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//7WsyChQLEftFiDOVTGkv3hFpyyt.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Avengers: Infinity War
        </div>
        <div
          className="ratingBox"
        >
          8.3
        </div>
        <div
          className="genresBox"
        >
          Adventure, Action, Fantasy
        </div>
        <div
          className="hidden"
        >
          As the Avengers and their allies have continued to protect the world from threats too large for any one hero to handle, a new danger has emerged from the cosmic shadows: Thanos. A despot of intergalactic infamy, his goal is to collect all six Infinity Stones, artifacts of unimaginable power, and use them to inflict his twisted will on all of reality. Everything the Avengers have fought for has led up to this moment - the fate of Earth and existence itself has never been more uncertain.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//fw02ONlDhrYjTSZV8XO6hhU3ds3.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Bumblebee
        </div>
        <div
          className="ratingBox"
        >
          6.5
        </div>
        <div
          className="genresBox"
        >
          Action, Adventure, Science Fiction
        </div>
        <div
          className="hidden"
        >
          On the run in the year 1987, Bumblebee finds refuge in a junkyard in a small Californian beach town. Charlie, on the cusp of turning 18 and trying to find her place in the world, discovers Bumblebee, battle-scarred and broken.  When Charlie revives him, she quickly learns this is no ordinary yellow VW bug.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//f27RTnWEc2bhVtfSlKmxxypznfR.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Widows
        </div>
        <div
          className="ratingBox"
        >
          6.7
        </div>
        <div
          className="genresBox"
        >
          Crime, Drama, Thriller
        </div>
        <div
          className="hidden"
        >
          A police shootout leaves four thieves dead during an explosive armed robbery attempt in Chicago. Their widows have nothing in common except a debt left behind by their spouses' criminal activities. Hoping to forge a future on their own terms, they join forces to pull off a heist.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//hgWAcic93phg4DOuQ8NrsgQWiqu.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Serenity
        </div>
        <div
          className="ratingBox"
        >
          5.1
        </div>
        <div
          className="genresBox"
        >
          Thriller, Drama
        </div>
        <div
          className="hidden"
        >
          Baker Dill is a fishing boat captain leading tours off a tranquil, tropical enclave called Plymouth Island. His quiet life is shattered, however, when his ex-wife Karen tracks him down with a desperate plea for help.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Venom
        </div>
        <div
          className="ratingBox"
        >
          6.6
        </div>
        <div
          className="genresBox"
        >
          Science Fiction, Action
        </div>
        <div
          className="hidden"
        >
          Investigative journalist Eddie Brock attempts a comeback following a scandal, but accidentally becomes the host of Venom, a violent, super powerful alien symbiote. Soon, he must rely on his newfound powers to protect the world from a shadowy organization looking for a symbiote of their own.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//hPuZR2bFoV0kuBrA98exWyvTf40.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Can You Ever Forgive Me?
        </div>
        <div
          className="ratingBox"
        >
          7.1
        </div>
        <div
          className="genresBox"
        >
          Drama, Comedy, Crime
        </div>
        <div
          className="hidden"
        >
          When bestselling celebrity biographer Lee Israel is no longer able to get published because she has fallen out of step with current tastes, she turns her art form to deception, abetted by her loyal friend Jack.  It is a story of two difficult, socially inept, people somehow becoming partners-in-crime in forgeries: one creating and the other selling it.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//t0jwlwFUrtkPJoq5JH9YrRVROXZ.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          Fate/stay night: Heavenâ€™s Feel II. lost butterfly
        </div>
        <div
          className="ratingBox"
        >
          3.9
        </div>
        <div
          className="genresBox"
        >
          Animation, Action, Fantasy
        </div>
        <div
          className="hidden"
        >
          Theatrical-release adaptation of the visual novel "Fate/stay night", following the third and final route. (Part 2 of a trilogy.)
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
    <div
      className="card"
    >
      <img
        alt="movie logo"
        className="imgWrapper"
        src="https://image.tmdb.org/t/p/w500//nJVPgIq3icZMQ26p5uWIFdUZJfL.jpg"
      />
      <div
        className="trailerBox"
      >
        <div
          className="circle"
        >
          <div
            className="triangle"
          />
        </div>
        <div
          className="label"
        >
          Watch Now
        </div>
        <button
          className="viewInfo common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          View Info
        </button>
      </div>
      <div
        className="back"
      >
        <div
          className="hidden"
        >
          <button
            className="close"
            onClick={[Function]}
            type="button"
          />
        </div>
        <div
          className="movieName"
        >
          LelleBelle
        </div>
        <div
          className="ratingBox"
        >
          4.4
        </div>
        <div
          className="genresBox"
        >
          Music, Romance
        </div>
        <div
          className="hidden"
        >
          19-year old Belle practices playing the violin diligently, but is unable to impart her music with a sensitive undertone. Her life changes when she discovers that sexual desire stimulates passion in her music.
        </div>
        <button
          className="hidden common"
          onClick={[Function]}
          style={Object {}}
          type="button"
        >
          Watch Now
        </button>
      </div>
    </div>
  </div>
  <div
    className="spinner"
  >
    <img
      alt="loading"
      src={
        Object {
          "process": [Function],
        }
      }
    />
  </div>
  <div
    className="loading"
  >
    LOADING
  </div>
</div>
`;
